iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Branch.java

```diff
@@ -57,16 +60,79 @@
 
     Side getSide(Terminal terminal);
 
-    CurrentLimits getCurrentLimits(Side side);
+    /**
+     * @deprecated Use {@link #getOperationalLimits1(LimitType, Class)} or {@link #getOperationalLimits2(LimitType, Class)} instead.
+     */
+    @Deprecated
+    default CurrentLimits getCurrentLimits(Side side) {
+        if (side == Side.ONE) {
+            return getOperationalLimits1(LimitType.CURRENT, CurrentLimits.class);
+        } else if (side == Side.TWO) {
+            return getOperationalLimits2(LimitType.CURRENT, CurrentLimits.class);
+        }
+        throw new AssertionError("Unexpected side: " + side);
+    }
 
-    CurrentLimits getCurrentLimits1();
+    /**
+     * @deprecated Use {@link #getOperationalLimits1(LimitType, Class)} instead.
+     */
+    @Deprecated
+    default CurrentLimits getCurrentLimits1() {
+        return getOperationalLimits1(LimitType.CURRENT, CurrentLimits.class);
+    }
+
+    default List<OperationalLimits> getOperationalLimits1() {
+        return Collections.singletonList(getCurrentLimits1());
+    }
+
+    default <L extends OperationalLimits> L getOperationalLimits1(LimitType limitType, Class<L> limitClazz) {
+        return limitType == LimitType.CURRENT && limitClazz == CurrentLimits.class ? (L) getCurrentLimits1() : null;
```

Comment:
I don't know how it's possible, but do we really need to pass both parameters? For instance, in `ReactiveLimitsHolder`, we only pass the class:
```
<L extends ReactiveLimits> L getReactiveLimits(Class<L> type)
```

Reply:
The issue is: what do we do if a user passes a parent class of two different kinds of limits? (for example, LoadingLimits.class)